ckanext/canada/plugins.py

```diff
@@ -407,6 +407,16 @@ def before_index(self, data_dict):
             for cr in data_dict['credit']:
                 cr.pop('__extras', None)
 
+        if data_dict.get('relationship'):
+            related_relationships = []
+            related_types = []
+            for rel in data_dict['relationship']:
+                related_relationships.append(rel.get('related_relationship'))
+                related_types.append(rel.get('related_type'))
+            data_dict['related_relationship'] = related_relationships
+            data_dict['related_type'] = related_types
```

Comment:
list comprehensions are shorter and run a little faster than for loops
```suggestion
            related_relationships = [rel['related_relationship'] for rel in data_dict['relationship']]
            related_types = [rel['related_type'] for rel in data_dict['relationship']]
```